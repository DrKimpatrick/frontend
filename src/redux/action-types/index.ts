// This is where we will add action types