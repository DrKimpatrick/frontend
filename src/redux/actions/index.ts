// This is where we will add our action creators
export {};
