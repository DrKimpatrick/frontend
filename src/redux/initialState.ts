import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
