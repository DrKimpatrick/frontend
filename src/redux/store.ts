import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import apiMiddleware from 'middleware/apiMiddleware';
import rootReducer from './reducers/rootReducer';
import initialState from './initialState';

const middleware = [apiMiddleware];

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
