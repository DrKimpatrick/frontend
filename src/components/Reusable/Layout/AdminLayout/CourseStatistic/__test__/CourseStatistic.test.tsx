import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Component from '../CourseStatistic';
import { Store } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';

let store: Store;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const initialState = {
  courses: {
    numberOfPendingCourse: 4,
    numberOfAcceptedCourse: 4,
    numberOfDeclinedCourse: 5
  }
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/path'
  })
}));
describe('Course Statistic', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Router>
        <Provider store={store}>
          <Component />
        </Provider>
      </Router>,
      div
    );
  });

  it('should create snapshot', () => {
    const tree = render(
      <Router>
        <Provider store={store}>
          <Component />
        </Provider>
      </Router>
    );

    expect(tree).toMatchSnapshot();
  });
});
