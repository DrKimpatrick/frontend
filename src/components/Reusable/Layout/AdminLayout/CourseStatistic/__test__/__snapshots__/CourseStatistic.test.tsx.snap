// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Statistic should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="courseStatistic w-full flex items-center"
      >
        <div
          class="export"
        >
          <button
            class="flex items-center"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
              />
            </svg>
            <h5>
              Export
            </h5>
          </button>
        </div>
        <div
          class="statistic"
        >
          <ul>
            <li>
              <a
                href="/accepted-course"
              >
                <small
                  style="color: rgb(161, 161, 161);"
                >
                  Accepted Courses:
                </small>
                <small
                  class="font-bold mx-1"
                  style="color: rgb(77, 155, 98);"
                >
                  4
                </small>
              </a>
            </li>
            <li>
              <a
                href="/pending-course"
              >
                <small
                  style="color: rgb(161, 161, 161);"
                >
                  Pending Courses:
                </small>
                <small
                  class="font-bold mx-1"
                  style="color: rgb(20, 55, 177);"
                >
                  4
                </small>
              </a>
            </li>
            <li>
              <a
                href="/declined-course"
              >
                <small
                  style="color: rgb(161, 161, 161);"
                >
                  Declined Courses:
                </small>
                <small
                  class="font-bold mx-1"
                  style="color: rgb(143, 37, 23);"
                >
                  5
                </small>
              </a>
            </li>
            <li>
              <button
                type="button"
              >
                Create Affiliate
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="courseStatistic w-full flex items-center"
    >
      <div
        class="export"
      >
        <button
          class="flex items-center"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
          <h5>
            Export
          </h5>
        </button>
      </div>
      <div
        class="statistic"
      >
        <ul>
          <li>
            <a
              href="/accepted-course"
            >
              <small
                style="color: rgb(161, 161, 161);"
              >
                Accepted Courses:
              </small>
              <small
                class="font-bold mx-1"
                style="color: rgb(77, 155, 98);"
              >
                4
              </small>
            </a>
          </li>
          <li>
            <a
              href="/pending-course"
            >
              <small
                style="color: rgb(161, 161, 161);"
              >
                Pending Courses:
              </small>
              <small
                class="font-bold mx-1"
                style="color: rgb(20, 55, 177);"
              >
                4
              </small>
            </a>
          </li>
          <li>
            <a
              href="/declined-course"
            >
              <small
                style="color: rgb(161, 161, 161);"
              >
                Declined Courses:
              </small>
              <small
                class="font-bold mx-1"
                style="color: rgb(143, 37, 23);"
              >
                5
              </small>
            </a>
          </li>
          <li>
            <button
              type="button"
            >
              Create Affiliate
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
